docs/intro.md

```diff
@@ -0,0 +1,1014 @@
+ +----------------------+
+```
+
+That's an efficient processing pattern because data is first reduced locally, which results in fewer records that need to be repartitioned by the `collect()`ion. Consider this less-than-recommended implementation for comparison:
+
+```python
+...
+
+# Use only global groupby - a less-than-recommended practice
+GB() \
+  .groupby(fname, counter) \
+  .run('person:*')
+
+# Expected result: the same, but slower :/
+```
+
+Using only our limited dataset it is unlikely that we'll be able to discern any difference in performance. Instead of collecting two records we'll be collecting three and that's hardly significant.
+
+Consider, however, what will happen once we start adding Revolio Clockberg Jr. (a.k.a Gearhead), Mr. Poopybutthole, Birdperson, Fart and the rest of the multiverse to the database. The number of input records and the number of families will increase accordingly, causing more and more records to be moved across the network and resulting in higher latencies.
+
+## Repartitioning Data
+!!! important "Important reminder"
+    Repartitioning impacts performance, so try avoiding it as much as possible.
+
+As mentioned earlier, when absolutely required, functions can repartition data in the cluster by using an arbitrary key. When data is repartitioned, each worker is assigned with a subset of the records' keys and these are shipped to it from all other workers.
+
+Let's make up a contrived example to demonstrate the inner workings. We'll add a requirement for storing  the families' head counts as simple strings in their respective String keys. Put differently, we expect that after running the function we'll be able to do this:
+
+!!! example "Example: expect results of retrieving family head counts
+    ````
+    127.0.0.1:30001> GET Smith
+    -> Redirected to slot [14205] located at 127.0.0.1:30003
+    "3"
+    127.0.0.1:30003> GET Sanchez
+    -> Redirected to slot [9503] located at 127.0.0.1:30002
+    "1"
+    127.0.0.1:30002> GET Pibbles
+    -> Redirected to slot [169] located at 127.0.0.1:30001
+    "1"
+    ````
+
+The trick in this case is ensuring that the target String keys we'll be using reside on the same shards as the workers that are writing them. The distribution of results should follow the cluster's partitioning scheme, just like that of the input records.
+
+To do that, we'll modify the function to include the [`repartition()`](operations.md#repartition) operation:
+
+```python
+def fname(x):
+  ''' Extracts the family name from a person's record '''
+  return x['value']['name'].split(' ')[1]
+
+def key(x):
+  ''' Extracts the key of a record '''
+  return x['key']
+
+def counter(k, a, r):
+  ''' Counts records '''
+  return (a if a else 0) + 1
+
+def summer(k, a, r):
+  ''' Sums record values '''
+  return (a if a else 0) + r['value']
+
+# Repartition for storing counts
+GB() \
+  .localgroupby(fname, counter) \
+  .repartition(key) \
+  .localgroupby(key, summer) \
```

Comment:
Maybe we should explain that aggregateby is doing exactly this.

Reply:
Fixed via 079f280